### PR TITLE
Feature/30 31 design controls for search doctors

### PR DESCRIPTION
**Pull request by TREN-D**

### CODE REVIEW

File: InternetHospitalWebUI/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (martyniukroman) [inline InternetHospitalWebUI/.vscode/launch.json:2]: Microsoft default comments

File: InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  templateUrl: './doctor-list-search-item.component.html',
+  styleUrls: ['./doctor-list-search-item.component.scss']
+})
+export class DoctorListSearchItemComponent implements OnInit {
+  @Input()
+  specializations: Specialization[];
+  @Output()
+  onSearch= new EventEmitter();
+
+  selectedSpecialization = "";
```
Comment (martyniukroman) [inline InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.ts:18]: better to use string.empty;

File: InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  ngOnInit() {
+  }
+
+  search() {
+    this.onSearch.emit({searchKey : this.searchKey,selectedSpecialization : this.selectedSpecialization});
+  }
+
+  onSearchClear() {
+    this.searchKey = "";
```
Comment (martyniukroman) [inline InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.ts:31]: better to use string.empty;

File: InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div class="container" fxLayout="column"  fxLayoutGap="0.5%" fxLayoutWrap fxLayoutAlign="center">
+  <mat-form-field fxFlex="40%" class="search-form-field" floatLabel="never" style="width:100%">
```
Comment (voznovych) [inline InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.html]: Put width in scss.

File: InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+export class DoctorListSearchItemComponent implements OnInit {
+  @Input()
+  specializations: Specialization[];
+  @Output()
+  onSearch= new EventEmitter();
+
+  selectedSpecialization = "";
+  searchKey: string;  
+
+  constructor(private service:DoctorsService, private paginationService: PaginationService) { }
```
Comment (voznovych) [inline InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-search-item/doctor-list-search-item.component.ts]: Put space after service: .

File: InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-item/doctor-list-item.component.html [hunk 6]
```diff
@@ -1,13 +1,13 @@
-<div class="container" fxLayout="row" fxLayout.xs="column" fxLayoutGap="0.5%" fxLayoutWrap fxLayoutAlign="center">
+<div class="container" fxLayout="row" fxLayoutGap="0.5%" fxLayoutWrap fxLayoutAlign="center">
   <mat-card style="width:100%">
```
Comment (voznovych) [inline InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-item/doctor-list-item.component.html]: Put width in scss.

File: InternetHospitalWebUI/src/app/Services/doctors.service.ts [hunk 7]
```diff
@@ -1,22 +1,58 @@
...
 
 @Injectable({
   providedIn: 'root'
 })
 export class DoctorsService {
   doctorsList: Doctor[];
+  specializations: Specialization[];
+  doctorsAmount: number;
+
   url = HOST_URL + '/api/Doctors';
```
Comment (voznovych) [inline InternetHospitalWebUI/src/app/Services/doctors.service.ts]: 'api/Doctors/' in const.

File: InternetHospitalWebUI/src/app/config.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-export const HOST_URL = 'https://localhost:44357';
+export const HOST_URL = 'https://localhost:5001';
```
Comment (edward0802) [inline InternetHospitalWebUI/src/app/config.ts]: fix variable names and value

File: InternetHospitalWebUI/src/app/Services/doctors.service.ts [hunk 9]
```diff
@@ -1,22 +1,58 @@
...
+    if (name != null && name != "") {
+      this.httpOptions.params =  this.httpOptions.params.set("searchbyname", name);
+    }
+    else {
+      this.httpOptions.params = this.httpOptions.params.delete("searchbyname");
+    }
 
-  constructor(private http: HttpClient) { }
+    if (specialization != 0 && !isNaN(specialization)) {
+      this.httpOptions.params =  this.httpOptions.params.set("searchbyspecialization", specialization.toString())
```
Comment (voznovych) [inline InternetHospitalWebUI/src/app/Services/doctors.service.ts]: Put "searchbyspecialization" in const.

Comment (voznovych) [reply InternetHospitalWebUI/src/app/Services/doctors.service.ts]: And another strings.

File: InternetHospitalWebUI/src/app/config.ts (hunk 8 above)
Comment (voznovych) [inline InternetHospitalWebUI/src/app/config.ts]: Whyyyyyyy?

File: InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-item/doctor-list-item.component.scss [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    max-width: 100%;
+    margin: 1px;
+    padding-top: 25px;
+}
+
+.mat-card-content {
+    margin-bottom: 0;
+    padding-left: 5px;
+    margin-top: 10px;
+    word-wrap:break-word;
```
Comment (voznovych) [inline InternetHospitalWebUI/src/app/Components/DoctorList/doctor-list/doctor-list-item/doctor-list-item.component.scss]: Put space after "word-wrap:".

File: InternetHospitalWebUI/src/app/Components/Layout/header/header.component.scss [hunk 11]
```diff
@@ -1,3 +1,8 @@
 .example-fill-remaining-space {
-    flex: 1 1 auto;
-  }
\ No newline at end of file
+  flex: 1 1 auto;
+}
+
+a {
+  font-size: 14px;
+  text-decoration:none;
```
Comment (voznovych) [inline InternetHospitalWebUI/src/app/Components/Layout/header/header.component.scss]: Put space after "text-decoration:".

**Review by voznovych — APPROVED** [review]

**Review by martyniukroman — APPROVED** [review]

**Review by edward0802 — APPROVED** [review]

**Pull request merged**